auth/range_perm_cache.go

```diff
@@ -112,30 +37,46 @@ func getMergedPerms(tx backend.BatchTx, userName string) *unifiedRangePermission
 		}
 
 		for _, perm := range role.KeyPermission {
-			rp := &rangePerm{begin: perm.Key, end: perm.RangeEnd}
-
-			switch perm.PermType {
-			case authpb.READWRITE:
-				readPerms = append(readPerms, rp)
-				writePerms = append(writePerms, rp)
-
-			case authpb.READ:
-				readPerms = append(readPerms, rp)
-
-			case authpb.WRITE:
-				writePerms = append(writePerms, rp)
+			if len(perm.RangeEnd) != 0 {
```

Comment:
Ah, yes, the type of the return value of `NewStringAffinePoint()` is also `Interval`, so it can be simplified as you pointed. I'll fix it.